inlong-agent/agent-installer/src/main/java/org/apache/inlong/agent/installer/ModuleManager.java

```diff
@@ -106,51 +120,110 @@ public String getCurrentMd5() {
         return currentMd5;
     }
 
+    public ModuleConfig getModule(Integer moduleId) {
+        return currentModules.get(moduleId);
+    }
+
     /**
-     * thread for core thread.
+     * Thread for core thread.
      *
      * @return runnable profile.
      */
     private Runnable coreThread() {
         return () -> {
-            Thread.currentThread().setName("task-manager-core");
+            Thread.currentThread().setName("module-manager-core");
+            restoreFromLocalFile(confPath);
             while (isRunnable()) {
                 try {
-                    AgentUtils.silenceSleepInMs(CORE_THREAD_SLEEP_TIME);
                     dealWithConfigQueue(configQueue);
                     AuditUtils.add(AuditUtils.AUDIT_ID_AGENT_TASK_MGR_HEARTBEAT, "", "",
                             AgentUtils.getCurrentTime(), 1, 1);
+                    AgentUtils.silenceSleepInMs(CORE_THREAD_SLEEP_TIME);
                 } catch (Throwable ex) {
                     LOGGER.error("exception caught", ex);
                     ThreadUtils.threadThrowableHandler(Thread.currentThread(), ex);
+                    AgentUtils.silenceSleepInMs(CORE_THREAD_SLEEP_TIME);
                 }
             }
         };
     }
 
+    public void restoreFromLocalFile(String confPath) {
+        LOGGER.info("restore modules from local file");
+        Reader reader = null;
+        try {
+            InputStream inputStream = new FileInputStream(confPath + LOCAL_CONFIG_FILE);
+            if (inputStream != null) {
+                reader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+                JsonElement tmpElement = JsonParser.parseReader(reader).getAsJsonObject();
+                ConfigResult curConfig = GSON.fromJson(tmpElement.getAsJsonObject(), ConfigResult.class);
+                if (curConfig.getMd5() != null && curConfig.getModuleList() != null) {
+                    currentMd5 = curConfig.getMd5();
+                    curConfig.getModuleList().forEach((module) -> {
+                        currentModules.put(module.getId(), module);
+                    });
+                } else {
+                    LOGGER.info("modules in local file invalid");
+                }
+            } else {
+                LOGGER.info("local module json file new FileInputStream failed");
+            }
+        } catch (FileNotFoundException e) {
+            LOGGER.info("local module json file not found");
```

Comment:
LOGGER.warn or LOGGER.error

Reply:
Even if there are no files, it is considered normal

Reply:
Got it! Thank you for your explanation.